src/inference/src/dev/threading/cpu_streams_executor.cpp

```diff
@@ -335,27 +334,75 @@ struct CPUStreamsExecutor::Impl {
     // will be counted by thread_local t_stream_count_map.
     // when the customer's thread is destoryed, the stream's count will became 1,
     // Call local() will reuse one of them, and release others.
+    // it's only a workaround for ticket CVS-111490, please be carefully when need to modify
+    // CustomeThreadLocal::local(), especially like operations that will affect the count of
+    // CustomThreadLocal::ThreadId
     class CustomThreadLocal : public ThreadLocal<std::shared_ptr<Stream>> {
+        class ThreadId {
+        public:
+            explicit ThreadId(std::thread::id& id)
+                : _id_ptr(new std::thread::id(id)),
+                  _count_ptr(new std::atomic_int(1)) {}
+            ~ThreadId() {
+                if (_count_ptr->fetch_sub(1) == 1) {
+                    delete _id_ptr;
+                    delete _count_ptr;
+                }
+            }
+            ThreadId* copy() {
```

Comment:
Rename to `clone` and return `std::shared_ptr` instead of raw pointer to avoid any memory leaks.

Reply:
May be name `fetch` would even more suitable.